content/asm_2.md

```diff
@@ -1,235 +1,508 @@
-    num2:   equ 50
-    ; initialize message
-    msg:    db "Sum is correct\n"
-
+    ;; The first number
+    num1 dq 0x64
+    ;; The second number
+    num2 dq 0x32
+    ;; The message to print if the sum is correct
+    msg  db "The sum is correct!", 10
+
+;; Definition of the .text section
 section .text
-
+    ;; Reference to the entry point of our program
     global _start
 
-;; entry point
+;; Entry point
 _start:
-    ; set num1's value to rax
-    mov rax, num1
-    ; set num2's value to rbx
-    mov rbx, num2
-    ; get sum of rax and rbx, and store it's value in rax
+    ;; Set the value of num1 to rax
+    mov rax, [num1]
+    ;; Set the value of num2 to rbx
+    mov rbx, [num2]
+    ;; Get the sum of rax and rbx. The result is stored in rax.
     add rax, rbx
-    ; compare rax and 150
+.compare:
+    ;; Compare the rax value with 150
     cmp rax, 150
-    ; go to .exit label if rax and 150 are not equal
+    ;; Go to the .exit label if the rax value is not 150
     jne .exit
-    ; go to .rightSum label if rax and 150 are equal
-    jmp .rightSum
-
-; Print message that sum is correct
-.rightSum:
-    ;; write syscall
-    mov     rax, 1
-    ;; file descritor, standard output
-    mov     rdi, 1
-    ;; message address
-    mov     rsi, msg
-    ;; length of message
-    mov     rdx, 15
-    ;; call write syscall
+    ;; Go to the .correctSum label if the rax value is 150
+    jmp .correctSum
+
+; Print message that the sum is correct
+.correctSum:
+    ;; Specify the system call number (1 is `sys_write`).
+    mov rax, 1
+    ;; Set the first argument of `sys_write` to 1 (`stdout`).
+    mov rdi, 1
+    ;; Set the second argument of `sys_write` to the reference of the `msg` variable.        
+    mov rsi, msg
+    ;; Set the third argument to the length of the `msg` variable's value (20 bytes).
+    mov rdx, 20
+    ;; Call the `sys_write` system call.
     syscall
-    ; exit from program
+    ; Go to the exit of the program.
     jmp .exit
 
 ; exit procedure
 .exit:
-    ; exit syscall
-    mov    rax, 60
-    ; exit code
-    mov    rdi, 0
-    ; call exit syscall
+    ;; Specify the number of the system call (60 is `sys_exit`).
+    mov rax, 60
+    ;; Set the first argument of `sys_exit` to 0. The 0 status code is success.
+    mov rdi, 0
+    ;; Call the `sys_exit` system call.
     syscall
 ```
 
-Let's go through the source code. First of all there is data section with two constants num1, num2 and variable msg with "Sum is correct\n" value. Now look at 14 line. There is begin of program's entry point. We transfer num1 and num2 values to general purpose registers rax and rbx. Sum it with add instruction. After execution of add instruction, it calculates sum of values from rax and rbx and store it's value to rax. Now we have sum of num1 and num2 in the rax register.
+First of all let's try to build, run our program with the similar commands that we have seen in the previous chapter and see the result:
```

Comment:
```suggestion
First, let's build and run our program using the commands we learned in the previous chapter to see the result:
```

Reply:
What do these commands do? Do they really both build and run the program? Or only build? Because below we have the command that actually runs the program:

> After we built our program, we can run it with:

Reply:
Yes, nice catch. Removed 'run' from the original sentence.